src/rosie.js

```diff
@@ -399,6 +399,17 @@ Factory.util = (function() {
         }
       }
       return dest;
+    },
+
+    /**
+     * Deep clones given object.
+     *
+     * @private
+     * @param {object} object
+     * @return {object}
+     */
+    deepClone: function(object) {
+      return JSON.parse(JSON.stringify(object));
```

Comment:
An alternative would be to use a library such as: https://lodash.com/docs#cloneDeep